internal/discovery/namespace_filter_test.go

```diff
@@ -44,7 +44,7 @@ func TestNamespaceFilterer_IsAllowed(t *testing.T) {
 				"newrelic.com/scrape": "true",
 			},
 			namespaceSelector: config.NamespaceSelector{
-				MatchLabels: map[string]string{
+				MatchLabels: map[string]interface{}{
```

Comment:
Is it really needed to support this? Kubernetes does not:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: test
  labels:
    test: test
    test2: test2
    algo: true
    algo2: 1
data:
  key: value
```
Gives this error:
```
% kubectl apply -f cm.yaml
error: unable to decode "cm.yaml": json: cannot unmarshal bool into Go struct field ObjectMeta.metadata.labels of type string
error: unable to decode "cm.yaml": json: cannot unmarshal number into Go struct field ObjectMeta.metadata.labels of type string
```

Reply:
Yes, that's true... we could directly fail. We agreed with @alvarocabanas to support it but maybe it doesn't make sense at all.

Reply:
For me as a user I would like to have it as we did, allowing me to bot put bool or string but if it's a kubernetes standard...

Reply:
(I hope I have enough context) 
Basically we are deciding to be more flexible allowing them to set:
```
    namespaceSelector: {}
    # If you want to include only namespaces with a given label you could do so by adding:
    # matchLabels:
    #   newrelic.com/scrape: true
```
That now it would be interpreted as `newrelic.com/scrape: "1"`

I get the point of @kang-makes and I agree with it. The only issue is that before it is very error prone. From my point of view could decide either to 
 - do what is implemented here knowing that we are being less strict than k8s it could be an issue at some point
 - fail with a nice message if we detect a bool/int
 - use the default parser adding an example with `"true"` and a comment explaining that `"` are needed 
 

 Maybe the last one is my favourite one, but I am fine with all of them